examples/ecs/generic_systems.rs

```diff
@@ -0,0 +1,21 @@
+use bevy::prelude::*;
+use bevy::ecs::component::Component;
+
+// When entities spawn in call an event
+
+struct A;
+struct B;
+
+fn spawn_entities<T: Component>(mut commands: Commands, q: Query<Entity, With<T>>) {
+    q.for_each(|entity| { commands.entity(entity).spawn();});
+    info!("Entity {:?} spawned in!")
```

Comment:
In rust format strings the `{}` implies a format argument and anything inside it are sort of format-argument-specifiers. 
`:?` means to print this this as `Debug`. 
[Here](https://doc.rust-lang.org/std/fmt/) are some good docs on learning about format specifiers.

`info!` needs a second argument to put in place of the `{:?}` (in this case the entity.
So this should be updated to

```rust
info!("Entity {:?} spawned in!", entity);
```

However also note, you are currently querying every `Entity` with a `T` component. 
This mean that these entities are already spawned, and `commands.entity(entity).spawn()` does not actually compile.
You can take a look at [these docs](https://docs.rs/bevy/0.5.0/bevy/ecs/prelude/struct.Commands.html) for better understanding of the `Commands` struct and how it operates.

Reply:
I agree with the proposed change :) I think it's a nice simple demonstration of this pattern and that specific use case (spawning entities with a specific component type) is very close to the sort of thing I find myself doing in practice.

That said, I have a nice simple example of this pattern to control dispatch to various buttons in my [Sudoku game](https://github.com/Leafwing-Studios/bevy-sudoku/blob/c872ddfa19179bbba38ba7cf64767c73fa033f12/src/ui.rs#L141). If you want something more interactive, I'd consider showing off that pattern in your example. The overall logic is:

1. Create buttons with marker components.
2. Insert one generic system instance per marker component type.
3. When those buttons are pressed, fire off an event of the marker component type to later be read.

Reply:
Hi, I had 2 questions,

1. I am pretty new to Rust but, for @NathanSWard refactored example is the '_' being used to ignore something?
2. For creating a more interactive example if I created a button that spawned entities and when clicked activated an event would that be acceptable?

Reply:
1. `_` is used to denote a variable that is not being used later :) `_foo` is also sometimes used in case you want to be more explicit about what the type is.

2. That sounds like it could work :) The important things are to make sure that a) each of your generic systems results in different behavior b) the user can observe these differences. Perhaps create buttons where each button pressed spawns a different type of entity that is visually distinct?